### ISSUE TITLE
Fix POSIX unistd sleep functions

### ISSUE THREAD

**Issue by ekrich** [scala-native/scala-native#2395]
`sleep` that should take unsigned and return unsigned - https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/unistd.h.htm

`usleep` is no longer in the specification, as described in this reference
POSIX.1-2001 declares this function obsolete; use nanosleep(2) instead. POSIX.1-2008 removes the specification of usleep().
When it comes to incorrect sleep return type I would suggest creating an issue, so we don't forget about that. We should change it in the next major release (0.5.x)

Also, this reference: https://newbedev.com/is-there-an-alternative-sleep-function-in-c-to-milliseconds

**Comment by LeeTibbert** [comment]
The opengroup link posted can no longer be found.  Everything that I have found indicates
that the return type is `int`, which is how SN posix `unistd.h` currently declares it:
```
def usleep(usecs: CUnsignedInt): CInt = extern

```
`man usleep` says: 
```
      #include <unistd.h>

       int usleep(useconds_t usec);

    4.3BSD,  POSIX.1-2001.   POSIX.1-2001  declares this function obsolete;
       use nanosleep(2) instead.  POSIX.1-2008 removes  the  specification  of
       usleep().

       On  the original BSD implementation, and in glibc before version 2.2.2,
       the return type of this function is void.  The  POSIX  version  returns
       int, and this is also the prototype used since glibc 2.2.2.

```
Re: removing the function

 SN posix lib is pretty loose/undefined about which standards it supports.
It tends to be 'big tent'.  

Since at least Linux "man" describes it, there is a case for removing it from
`posixlib` and creating a new `clib unistd.h` which contained it.  That would
leave the `clib` instance pretty rump/useless.

I propose that we leave it in place and add a `@deprecated` annotation,
saying removed from Posix and giving  the recommended nanosleep()
alternative.  In theory, that could be done for 0.4.5.

My overarching concern is the ration of benefit to effort.  `unistd.scala`
is missing _a_lot_ of Posix. Balancing that thought is that we should
make economical improvements when and as we can.

Do you concur with going with an annotation, setting up removal in 0.5.0 or 0.6.0?
After this amount of time, what is your current thinking on `unistd.h`?  (besides
making hex signs)

**Comment by LeeTibbert** [comment]
To follow Scala practice of yocto-minimal comments, if any, I put a one line
comment in  Posix `unistd.scala` referencing this entry.

I can not seem to shake my commercial Software Engineering background.
Someday this issue is going to recur and someone, especially the people
paying the bills, will be happy to have a clue.
```
/* Developer Note:                                                            
   *    usleep() was declared obsolete in POSIX.1-2001 and removed in           
   *    POSIX.1-2008. However, usleep() appears in linux glibc, macOS,          
   *    FreeBSD, and probably other BSD systems.                                
   *                                                                            
   *    After the deprecation interval passes, usleep() could be                
   *    moved to Scala Native libc. That creates the need to create             
   *    and populate a libc unistd.scala.                                       
   *                                                                            
   *    Given limited developer time, there is no urgency to removing           
   *    this usleep() now that it has been deprecated.                          
   */
```

**Comment by LeeTibbert** [comment]
Eric,

Tidying up:
1) After the PR is merged, I can create a new issue with the 0.5.0 milestone
    stating the POSIX  `usleep` should be removed.  That will help keep me
    from having to carry the task around in my head.

2) Can the 0.5.0 milestone be removed from the this issue?  I think the
     PR is perfectly appropriate for mainline and need not wait for 0.5.0.
     Since you added the label, I thought I would ask you about its
     removal.


Your thoughts?
